### DISCUSSION TITLE
EAP Identity problem

### DISCUSSION THREAD

**Discussion by drcyberg** [strongswan/strongswan#1817] [Q&A] [answered]
Hi @tobiasbrunner !

I have one EAP identity problem: I have 2 users, the authentication of the first user is checked according to EAP-RADIUS (everything works fine), while the other user is solved locally, i.e. in `ipsec.secrets` file `test : EAP 123456` would happen.

Mistake: If I enter the username and password for the first time, it doesn't work, but if I enter the same username and password again, it works.

It's like it first queried the radius server, then on the second attempt it queried the local file (`ipsec.secrets`).
I am correct?

I tried to change the sequence within the configuration, but it didn't work either (`eap-mschapv2` after `eap-radius`).

Here is the log:

<details><summary>Details</summary>
<p>

root@vpn:/var/log# cat syslog
Jul 28 08:38:43 vpn charon: 12[NET] received packet: from 185.27.62.231[500] to 193.188.192.136[500] (632 bytes)
Jul 28 08:38:43 vpn charon: 12[ENC] parsed IKE_SA_INIT request 0 [ SA KE No N(FRAG_SUP) N(NATD_S_IP) N(NATD_D_IP) V V V V ]
Jul 28 08:38:43 vpn charon: 12[IKE] received MS NT5 ISAKMPOAKLEY v9 vendor ID
Jul 28 08:38:43 vpn charon: 12[IKE] received MS-Negotiation Discovery Capable vendor ID
Jul 28 08:38:43 vpn charon: 12[IKE] received Vid-Initial-Contact vendor ID
Jul 28 08:38:43 vpn charon: 12[ENC] received unknown vendor ID: 01:52:8b:bb:c0:06:96:12:18:49:ab:9a:1c:5b:2a:51:00:00:00:02
Jul 28 08:38:43 vpn charon: 12[IKE] 185.27.62.231 is initiating an IKE_SA
Jul 28 08:38:43 vpn charon: 12[IKE] remote host is behind NAT
Jul 28 08:38:43 vpn charon: 12[ENC] generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(CHDLESS_SUP) ]
Jul 28 08:38:43 vpn charon: 12[NET] sending packet: from 193.188.192.136[500] to 185.27.62.231[500] (440 bytes)
Jul 28 08:38:43 vpn charon: 16[NET] received packet: from 185.27.62.231[4500] to 193.188.192.136[4500] (1756 bytes)
Jul 28 08:38:43 vpn charon: 16[ENC] parsed IKE_AUTH request 1 [ IDi CERTREQ N(MOBIKE_SUP) CPRQ(ADDR DNS NBNS SRV ADDR6 DNS6 SRV6) SA TSi TSr ]
Jul 28 08:38:43 vpn charon: 16[IKE] received cert request for "CN=CP-VPN root CA"
Jul 28 08:38:43 vpn charon: 16[IKE] received 72 cert requests for an unknown ca
Jul 28 08:38:43 vpn charon: 16[IKE] initiating EAP_MSCHAPV2 method (id 0xB8)
Jul 28 08:38:43 vpn charon: 16[IKE] peer supports MOBIKE, but disabled in config
Jul 28 08:38:43 vpn charon: 16[IKE] authentication of 'vpn.cyberpointer.net' (myself) with RSA signature successful
Jul 28 08:38:43 vpn charon: 16[IKE] sending end entity cert "CN=193.188.192.136"
Jul 28 08:38:43 vpn charon: 16[ENC] generating IKE_AUTH response 1 [ IDr CERT AUTH EAP/REQ/MSCHAPV2 ]
Jul 28 08:38:43 vpn charon: 16[NET] sending packet: from 193.188.192.136[4500] to 185.27.62.231[4500] (1980 bytes)
Jul 28 08:38:43 vpn charon: 05[NET] received packet: from 185.27.62.231[4500] to 193.188.192.136[4500] (140 bytes)
Jul 28 08:38:43 vpn charon: 05[ENC] parsed IKE_AUTH request 2 [ EAP/RES/MSCHAPV2 ]
Jul 28 08:38:43 vpn charon: 05[IKE] EAP-MS-CHAPv2 username: '%any'
Jul 28 08:38:43 vpn charon: 05[IKE] no EAP key found for hosts '%any' - '%any'
Jul 28 08:38:43 vpn charon: 05[IKE] EAP-MS-CHAPv2 verification failed, retry (1)
Jul 28 08:38:44 vpn charon: 16[MGR] ignoring request with ID 2, already processing
Jul 28 08:38:45 vpn charon: 05[ENC] generating IKE_AUTH response 2 [ EAP/REQ/MSCHAPV2 ]
Jul 28 08:38:45 vpn charon: 05[NET] sending packet: from 193.188.192.136[4500] to 185.27.62.231[4500] (124 bytes)
Jul 28 08:38:45 vpn charon: 13[NET] received packet: from 185.27.62.231[4500] to 193.188.192.136[4500] (140 bytes)
Jul 28 08:38:45 vpn charon: 13[ENC] parsed IKE_AUTH request 2 [ EAP/RES/MSCHAPV2 ]
Jul 28 08:38:45 vpn charon: 13[IKE] received retransmit of request with ID 2, retransmitting response
Jul 28 08:38:45 vpn charon: 13[NET] sending packet: from 193.188.192.136[4500] to 185.27.62.231[4500] (124 bytes)
Jul 28 08:38:49 vpn charon: 16[NET] received packet: from 185.27.62.231[4500] to 193.188.192.136[4500] (140 bytes)
Jul 28 08:38:49 vpn charon: 16[ENC] parsed IKE_AUTH request 3 [ EAP/RES/MSCHAPV2 ]
Jul 28 08:38:49 vpn charon: 16[IKE] EAP-MS-CHAPv2 username: 'test'
Jul 28 08:38:49 vpn charon: 16[ENC] generating IKE_AUTH response 3 [ EAP/REQ/MSCHAPV2 ]
Jul 28 08:38:49 vpn charon: 16[NET] sending packet: from 193.188.192.136[4500] to 185.27.62.231[4500] (140 bytes)
Jul 28 08:38:49 vpn ipsec[11042]: 11[IKE] scheduling reauthentication in 3313s
Jul 28 08:38:49 vpn ipsec[11042]: 11[IKE] maximum IKE_SA lifetime 3493s
Jul 28 08:38:49 vpn ipsec[11042]: 11[IKE] peer requested virtual IP %any
Jul 28 08:38:49 vpn ipsec[11042]: 11[IKE] assigning virtual IP 11.11.11.1 to peer 'test'
Jul 28 08:38:49 vpn ipsec[11042]: 11[IKE] peer requested virtual IP %any6
Jul 28 08:38:49 vpn ipsec[11042]: 11[IKE] no virtual IP found for %any6 requested by 'test'
Jul 28 08:38:49 vpn ipsec[11042]: 11[IKE] CHILD_SA test-cp-vpn{1} established with SPIs cd26a53a_i 6b77b786_o and TS 11.11.11.0/24 === 11.11.11.0/24
Jul 28 08:38:49 vpn ipsec[11042]: 11[ENC] generating IKE_AUTH response 5 [ AUTH CPRP(ADDR) SA TSi TSr N(AUTH_LFT) ]
Jul 28 08:38:49 vpn ipsec[11042]: 11[NET] sending packet: from 193.188.192.136[4500] to 185.27.62.231[4500] (204 bytes)
Jul 28 08:38:49 vpn ipsec[11042]: 08[NET] received packet: from 185.27.62.231[4500] to 193.188.192.136[4500] (76 bytes)
Jul 28 08:38:49 vpn ipsec[11042]: 08[ENC] parsed INFORMATIONAL request 6 [ D ]
Jul 28 08:38:49 vpn ipsec[11042]: 08[IKE] received DELETE for ESP CHILD_SA with SPI 6b77b786
Jul 28 08:38:49 vpn ipsec[11042]: 08[IKE] closing CHILD_SA test-cp-vpn{1} with SPIs cd26a53a_i (0 bytes) 6b77b786_o (0 bytes) and TS 11.11.11.0/24 === 11.11.11.0/24
Jul 28 08:38:49 vpn ipsec[11042]: 08[IKE] sending DELETE for ESP CHILD_SA with SPI cd26a53a
Jul 28 08:38:49 vpn ipsec[11042]: 08[IKE] CHILD_SA closed
Jul 28 08:38:49 vpn ipsec[11042]: 08[ENC] generating INFORMATIONAL response 6 [ D ]
Jul 28 08:38:49 vpn ipsec[11042]: 08[NET] sending packet: from 193.188.192.136[4500] to 185.27.62.231[4500] (76 bytes)
Jul 28 08:38:49 vpn ipsec[11042]: 11[NET] received packet: from 185.27.62.231[4500] to 193.188.192.136[4500] (76 bytes)
Jul 28 08:38:49 vpn ipsec[11042]: 11[ENC] parsed INFORMATIONAL request 7 [ D ]
Jul 28 08:38:49 vpn ipsec[11042]: 11[IKE] received DELETE for IKE_SA test-cp-vpn[1]
Jul 28 08:38:49 vpn ipsec[11042]: 11[IKE] deleting IKE_SA test-cp-vpn[1] between 193.188.192.136[vpn.cyberpointer.net]...185.27.62.231[192.168.1.108]
Jul 28 08:38:49 vpn ipsec[11042]: 11[IKE] IKE_SA deleted
Jul 28 08:38:49 vpn ipsec[11042]: 11[ENC] generating INFORMATIONAL response 7 [ ]
Jul 28 08:38:49 vpn ipsec[11042]: 11[NET] sending packet: from 193.188.192.136[4500] to 185.27.62.231[4500] (76 bytes)
Jul 28 08:38:49 vpn ipsec[11042]: 12[NET] received packet: from 185.27.62.231[500] to 193.188.192.136[500] (632 bytes)
Jul 28 08:38:49 vpn ipsec[11042]: 12[ENC] parsed IKE_SA_INIT request 0 [ SA KE No N(FRAG_SUP) N(NATD_S_IP) N(NATD_D_IP) V V V V ]
Jul 28 08:38:49 vpn ipsec[11042]: 12[IKE] received MS NT5 ISAKMPOAKLEY v9 vendor ID
Jul 28 08:38:49 vpn ipsec[11042]: 12[IKE] received MS-Negotiation Discovery Capable vendor ID
Jul 28 08:38:49 vpn ipsec[11042]: 12[IKE] received Vid-Initial-Contact vendor ID
Jul 28 08:38:49 vpn ipsec[11042]: 12[ENC] received unknown vendor ID: 01:52:8b:bb:c0:06:96:12:18:49:ab:9a:1c:5b:2a:51:00:00:00:02
Jul 28 08:38:49 vpn ipsec[11042]: 12[IKE] 185.27.62.231 is initiating an IKE_SA
Jul 28 08:38:49 vpn ipsec[11042]: 12[IKE] remote host is behind NAT
Jul 28 08:38:49 vpn ipsec[11042]: 12[ENC] generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(CHDLESS_SUP) ]
Jul 28 08:38:49 vpn ipsec[11042]: 12[NET] sending packet: from 193.188.192.136[500] to 185.27.62.231[500] (440 bytes)
Jul 28 08:38:49 vpn ipsec[11042]: 16[NET] received packet: from 185.27.62.231[4500] to 193.188.192.136[4500] (1756 bytes)
Jul 28 08:38:49 vpn ipsec[11042]: 16[ENC] parsed IKE_AUTH request 1 [ IDi CERTREQ N(MOBIKE_SUP) CPRQ(ADDR DNS NBNS SRV ADDR6 DNS6 SRV6) SA TSi TSr ]
Jul 28 08:38:49 vpn ipsec[11042]: 16[IKE] received cert request for "CN=CP-VPN root CA"
Jul 28 08:38:49 vpn ipsec[11042]: 16[IKE] received 72 cert requests for an unknown ca
Jul 28 08:38:49 vpn ipsec[11042]: 16[IKE] initiating EAP_MSCHAPV2 method (id 0xB8)
Jul 28 08:38:49 vpn ipsec[11042]: 16[IKE] peer supports MOBIKE, but disabled in config
Jul 28 08:38:49 vpn ipsec[11042]: 16[IKE] authentication of 'vpn.cyberpointer.net' (myself) with RSA signature successful
Jul 28 08:38:49 vpn ipsec[11042]: 16[IKE] sending end entity cert "CN=193.188.192.136"
Jul 28 08:38:49 vpn ipsec[11042]: 16[ENC] generating IKE_AUTH response 1 [ IDr CERT AUTH EAP/REQ/MSCHAPV2 ]
Jul 28 08:38:49 vpn ipsec[11042]: 16[NET] sending packet: from 193.188.192.136[4500] to 185.27.62.231[4500] (1980 bytes)
Jul 28 08:38:49 vpn ipsec[11042]: 05[NET] received packet: from 185.27.62.231[4500] to 193.188.192.136[4500] (140 bytes)
Jul 28 08:38:49 vpn ipsec[11042]: 05[ENC] parsed IKE_AUTH request 2 [ EAP/RES/MSCHAPV2 ]
Jul 28 08:38:49 vpn ipsec[11042]: 05[IKE] EAP-MS-CHAPv2 username: '%any'
Jul 28 08:38:49 vpn ipsec[11042]: 05[IKE] no EAP key found for hosts '%any' - '%any'
Jul 28 08:38:49 vpn ipsec[11042]: 05[IKE] EAP-MS-CHAPv2 verification failed, retry (1)
Jul 28 08:38:49 vpn ipsec[11042]: 16[MGR] ignoring request with ID 2, already processing
Jul 28 08:38:49 vpn ipsec[11042]: 05[ENC] generating IKE_AUTH response 2 [ EAP/REQ/MSCHAPV2 ]
Jul 28 08:38:49 vpn ipsec[11042]: 05[NET] sending packet: from 193.188.192.136[4500] to 185.27.62.231[4500] (124 bytes)
Jul 28 08:38:49 vpn ipsec[11042]: 13[NET] received packet: from 185.27.62.231[4500] to 193.188.192.136[4500] (140 bytes)
Jul 28 08:38:49 vpn ipsec[11042]: 13[ENC] parsed IKE_AUTH request 2 [ EAP/RES/MSCHAPV2 ]
Jul 28 08:38:49 vpn ipsec[11042]: 13[IKE] received retransmit of request with ID 2, retransmitting response
Jul 28 08:38:49 vpn ipsec[11042]: 13[NET] sending packet: from 193.188.192.136[4500] to 185.27.62.231[4500] (124 bytes)
Jul 28 08:38:49 vpn ipsec[11042]: 16[NET] received packet: from 185.27.62.231[4500] to 193.188.192.136[4500] (140 bytes)
Jul 28 08:38:49 vpn ipsec[11042]: 16[ENC] parsed IKE_AUTH request 3 [ EAP/RES/MSCHAPV2 ]
Jul 28 08:38:49 vpn ipsec[11042]: 16[IKE] EAP-MS-CHAPv2 username: 'test'
Jul 28 08:38:49 vpn ipsec[11042]: 16[ENC] generating IKE_AUTH response 3 [ EAP/REQ/MSCHAPV2 ]
Jul 28 08:38:49 vpn ipsec[11042]: 16[NET] sending packet: from 193.188.192.136[4500] to 185.27.62.231[4500] (140 bytes)
Jul 28 08:38:49 vpn charon: 14[NET] received packet: from 185.27.62.231[4500] to 193.188.192.136[4500] (76 bytes)
Jul 28 08:38:49 vpn charon: 14[ENC] parsed IKE_AUTH request 4 [ EAP/RES/MSCHAPV2 ]
Jul 28 08:38:49 vpn charon: 14[IKE] EAP method EAP_MSCHAPV2 succeeded, MSK established
Jul 28 08:38:49 vpn charon: 14[ENC] generating IKE_AUTH response 4 [ EAP/SUCC ]
Jul 28 08:38:49 vpn charon: 14[NET] sending packet: from 193.188.192.136[4500] to 185.27.62.231[4500] (76 bytes)
Jul 28 08:38:49 vpn charon: 08[NET] received packet: from 185.27.62.231[4500] to 193.188.192.136[4500] (92 bytes)
Jul 28 08:38:49 vpn charon: 08[ENC] parsed IKE_AUTH request 5 [ AUTH ]
Jul 28 08:38:49 vpn charon: 08[IKE] authentication of '192.168.1.108' with EAP successful
Jul 28 08:38:49 vpn charon: 08[IKE] authentication of 'vpn.cyberpointer.net' (myself) with EAP
Jul 28 08:38:49 vpn charon: 08[IKE] IKE_SA test-cp-vpn[2] established between 193.188.192.136[vpn.cyberpointer.net]...185.27.62.231[192.168.1.108]
Jul 28 08:38:49 vpn charon: 08[IKE] scheduling reauthentication in 3302s
Jul 28 08:38:49 vpn charon: 08[IKE] maximum IKE_SA lifetime 3482s
Jul 28 08:38:49 vpn charon: 08[IKE] peer requested virtual IP %any
Jul 28 08:38:49 vpn charon: 08[IKE] assigning virtual IP 11.11.11.1 to peer 'test'
Jul 28 08:38:49 vpn charon: 08[IKE] peer requested virtual IP %any6
Jul 28 08:38:49 vpn charon: 08[IKE] no virtual IP found for %any6 requested by 'test'
Jul 28 08:38:49 vpn charon: 08[IKE] CHILD_SA test-cp-vpn{2} established with SPIs c1626a6d_i 20820716_o and TS 11.11.11.0/24 === 11.11.11.0/24
Jul 28 08:38:49 vpn vpn: + 192.168.1.108 11.11.11.0/24 == 185.27.62.231 -- 193.188.192.136 == 11.11.11.0/24
Jul 28 08:38:49 vpn charon: 08[ENC] generating IKE_AUTH response 5 [ AUTH CPRP(ADDR) SA TSi TSr N(AUTH_LFT) ]
Jul 28 08:38:49 vpn charon: 08[NET] sending packet: from 193.188.192.136[4500] to 185.27.62.231[4500] (204 bytes)
Jul 28 08:38:52 vpn charon: 06[NET] received packet: from 185.27.62.231[4500] to 193.188.192.136[4500] (76 bytes)
Jul 28 08:38:52 vpn charon: 06[ENC] parsed INFORMATIONAL request 6 [ D ]
Jul 28 08:38:52 vpn charon: 06[IKE] received DELETE for ESP CHILD_SA with SPI 20820716
Jul 28 08:38:52 vpn charon: 06[IKE] closing CHILD_SA test-cp-vpn{2} with SPIs c1626a6d_i (0 bytes) 20820716_o (0 bytes) and TS 11.11.11.0/24 === 11.11.11.0/24
Jul 28 08:38:52 vpn charon: 06[IKE] sending DELETE for ESP CHILD_SA with SPI c1626a6d
Jul 28 08:38:52 vpn charon: 06[IKE] CHILD_SA closed
Jul 28 08:38:52 vpn vpn: - 192.168.1.108 11.11.11.0/24 == 185.27.62.231 -- 193.188.192.136 == 11.11.11.0/24
Jul 28 08:38:52 vpn charon: 06[ENC] generating INFORMATIONAL response 6 [ D ]
Jul 28 08:38:52 vpn charon: 06[NET] sending packet: from 193.188.192.136[4500] to 185.27.62.231[4500] (76 bytes)
Jul 28 08:38:52 vpn charon: 10[NET] received packet: from 185.27.62.231[4500] to 193.188.192.136[4500] (76 bytes)
Jul 28 08:38:52 vpn charon: 10[ENC] parsed INFORMATIONAL request 7 [ D ]
Jul 28 08:38:52 vpn charon: 10[IKE] received DELETE for IKE_SA test-cp-vpn[2]
Jul 28 08:38:52 vpn charon: 10[IKE] deleting IKE_SA test-cp-vpn[2] between 193.188.192.136[vpn.cyberpointer.net]...185.27.62.231[192.168.1.108]
Jul 28 08:38:52 vpn charon: 10[IKE] IKE_SA deleted
Jul 28 08:38:52 vpn charon: 10[ENC] generating INFORMATIONAL response 7 [ ]
Jul 28 08:38:52 vpn charon: 10[NET] sending packet: from 193.188.192.136[4500] to 185.27.62.231[4500] (76 bytes)

</p>
</details> 

Thank you!

Regards: DrCyberg

**Comment by tobiasbrunner** [comment, marked as answer]
There is no EAP-Identity exchange. Make sure the _eap-identity_ plugin is loaded and configure `eap_id = %any` to request the identity before the EAP-MSCHAPv2 exchange is initiated.

**Reply by drcyberg** [reply]
Yes, the `eap-identity.conf` is loaded (`load = yes`) and I changed `eap_identity = %identity` to `%any` works fine. :)
Thank you Bro! Magical!
